scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -1328,70 +1327,102 @@
+                 (d/db conn) gpu-jobs user->quota user->usage num-considerable (atom {}) nil)))))
     (testing "jobs inside usage quota limited by num-considerable of 2"
       (let [user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}}
             user->quota {test-user {:count 10, :cpus 50, :mem 32768, :gpus 10}}
             num-considerable 2]
         (is (= [job-1 job-2]
                (sched/pending-jobs->considerable-jobs
-                 (d/db conn) non-gpu-jobs user->quota user->usage num-considerable nil)))
+                 (d/db conn) non-gpu-jobs user->quota user->usage num-considerable (atom {}) nil)))
         (is (= gpu-jobs
                (sched/pending-jobs->considerable-jobs
-                 (d/db conn) gpu-jobs user->quota user->usage num-considerable nil)))))
+                 (d/db conn) gpu-jobs user->quota user->usage num-considerable (atom {}) nil)))))
     (testing "jobs inside usage quota limited by num-considerable of 1"
       (let [user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}}
             user->quota {test-user {:count 10, :cpus 50, :mem 32768, :gpus 10}}
             num-considerable 1]
         (is (= [job-1]
                (sched/pending-jobs->considerable-jobs
-                 (d/db conn) non-gpu-jobs user->quota user->usage num-considerable nil)))
+                 (d/db conn) non-gpu-jobs user->quota user->usage num-considerable (atom {}) nil)))
         (is (= [job-5]
                (sched/pending-jobs->considerable-jobs
-                 (d/db conn) gpu-jobs user->quota user->usage num-considerable nil)))))
+                 (d/db conn) gpu-jobs user->quota user->usage num-considerable (atom {}) nil)))))
+    (testing "jobs inside usage quota influenced by optimizer"
+      (let [user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}}
+            user->quota {test-user {:count 10, :cpus 50, :mem 32768, :gpus 10}}
+            num-considerable 5
+            pool-name->optimizer-suggested-job-ids-atom (atom {"cpu-pool" [(:job/uuid job-4) (:job/uuid job-2) (:job/uuid job-6)]
+                                                              "gpu-pool" [(:job/uuid job-2) (:job/uuid job-6)]})]
+        (is (= [job-4 job-2 job-1 job-3]
+               (sched/pending-jobs->considerable-jobs
+                 (d/db conn) non-gpu-jobs user->quota user->usage num-considerable
+                 pool-name->optimizer-suggested-job-ids-atom "cpu-pool")))
+        (is (= [job-6 job-5]
+               (sched/pending-jobs->considerable-jobs
+                 (d/db conn) gpu-jobs user->quota user->usage num-considerable
+                 pool-name->optimizer-suggested-job-ids-atom "gpu-pool")))
+        (is (= {"cpu-pool" [] "gpu-pool" []}
+               @pool-name->optimizer-suggested-job-ids-atom))))
+    (testing "jobs inside usage quota influenced by optimizer limited by num-considerable of 1"
```

Comment:
I would favor a newline before the `testing` clause

Reply:
Done.